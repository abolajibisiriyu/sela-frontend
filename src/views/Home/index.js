import React from "react";
// import Layout from "../../blocks/Layout";
import HomeLayout from "../../components/HomeLayout";
import Header from "./components/Header";

function Home(props) {
  return (
    <HomeLayout>
      <Header />
      <h1>Home</h1>
    </HomeLayout>
  );
}

export default Home;
