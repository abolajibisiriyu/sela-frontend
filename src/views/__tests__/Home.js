import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import promise from "redux-promise-middleware";

import Home from "../Home";

const middlewares = [thunk, promise];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ projects: {} });

describe("Home Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
