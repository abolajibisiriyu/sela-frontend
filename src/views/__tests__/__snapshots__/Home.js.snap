// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  className="sc-bxivhb dvXZAB"
>
  <header
    className="sc-gzVnrw daRvCo"
  >
    <nav
      className="sc-dnqmqq gQTnSb"
    >
      <a
        className="sc-htoDjs jxoWAa"
        href="#"
      >
        SELA
      </a>
    </nav>
    <div
      className="sc-bZQynM fYKTQZ"
    >
      <h4
        className="big-size sc-iwsKbI geApFa"
      >
        Keep track of development 
        <br />
         projects near you
      </h4>
      <p
        className="big-size sc-gZMcBi cwhwfC"
      >
        The Sela Platform enables stakeholders share real time information
         
        <br />
        on the state of infrastructure projects to ensure transparent delivery
      </p>
      <a
        className="sc-EHOje eavTOz"
        onClick={[Function]}
      >
        <span>
          SEE PROJECTS
        </span>
        <span>
          <img
            alt="scroll down"
            src="expand-icon.svg"
          />
        </span>
      </a>
    </div>
  </header>
  <div
    className="sc-jWBwVP fDrbnp"
  >
    <div
      className="sc-cvbbAY hSWAFw"
    >
      <div
        className="title"
      >
        Search Projects By Location
      </div>
      <div
        className="input"
      >
        <svg
          color="rgba(79, 79, 79, 0.8)"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "rgba(79, 79, 79, 0.8)",
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          />
        </svg>
        <input
          placeholder="Ikorodu, Lagos"
          type="text"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </div>
  <div>
    <section
      className="sc-fjdhpX flNxrH"
    >
      <div
        className="sc-gqjmRU ftPiGr"
      >
        <div
          className="sc-VigVT wAawY"
        >
          <h4>
            All Projects
          </h4>
        </div>
        <div
          className="sc-chPdSV DIMEh"
        >
          <div
            className="sc-jzJRlG jVBuMd"
          >
            <h3>
              Sort by:
            </h3>
          </div>
          <div
            className="sc-kAzzGY cLAYFq"
          >
            <select
              className="sc-cSHVUG hIYkaR"
              onChange={[Function]}
            >
              <option
                value=""
              >
                None
              </option>
              <option
                value="budget"
              >
                Budget
              </option>
              <option
                value="startDate"
              >
                Start date
              </option>
              <option
                value="endDate"
              >
                End Date
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="sc-jTzLTM jroYUT"
      />
    </section>
  </div>
  <div
    className="sc-ifAKCX dihmFK"
  >
    (c) Sela. 
    2019
  </div>
</div>
`;
